inc/classes/Addons/Cloudflare/Cloudflare.php

```diff
@@ -0,0 +1,611 @@
+<?php
+namespace WP_Rocket\Addons\Cloudflare;
+
+use WP_Rocket\Admin\Options_Data;
+
+/**
+ * Cloudflare
+ *
+ * @since 3.5
+ * @author Soponar Cristina
+ */
+class Cloudflare {
+	/**
+	 * WP Rocket options instance.
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * Cloudflare API instance or WP_Error
+	 *
+	 * @var \Cloudflare\Api
+	 */
+	private $cloudflare_api;
+
+	/**
+	 * Cloudflare User Agent
+	 */
+	const CF_USER_AGENT = 'wp-rocket/' . WP_ROCKET_VERSION;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data $options WP Rocket options instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options        = $options;
+		$this->cloudflare_api = $this->get_cloudflare_instance();
+	}
+
+	/**
+	 * Get a Cloudflare\Api instance & the zone_id corresponding to the domain
+	 *
+	 * @since 2.8.21 Get the zone ID from the options
+	 * @since 2.8.18 Add try/catch to prevent fatal error Uncaugh Exception
+	 * @since 2.8.16 Update to Cloudflare API v4
+	 * @since 2.5
+	 *
+	 * @return Object Cloudflare instance & zone_id if credentials are correct, WP_Error otherwise
+	 */
+	public function get_cloudflare_instance() {
+		$cf_email             = $this->options->get( 'cloudflare_email', null );
+		$cf_api_key           = defined( 'WP_ROCKET_CF_API_KEY' ) ? WP_ROCKET_CF_API_KEY : $this->options->get( 'cloudflare_api_key', null );
+		$cf_zone_id           = $this->options->get( 'cloudflare_zone_id', null );
+		$is_api_keys_valid_cf = get_transient( 'rocket_cloudflare_is_api_keys_valid' );
+
+		if ( false === $is_api_keys_valid_cf ) {
+			$is_api_keys_valid_cf = $this->is_api_keys_valid( $cf_email, $cf_api_key, $cf_zone_id );
+			set_transient( 'rocket_cloudflare_is_api_keys_valid', $is_api_keys_valid_cf, 2 * WEEK_IN_SECONDS );
+		}
+
+		if ( is_wp_error( $is_api_keys_valid_cf ) ) {
+			return $is_api_keys_valid_cf;
+		}
+
+		$cf_api_instance = new \Cloudflare\Api( $cf_email, $cf_api_key );
+		$cf_api_instance->setCurlOption( CURLOPT_USERAGENT, self::CF_USER_AGENT );
```

Comment:
Smart idea to use that to pass the custom user agent, did you check that it's working correctly with the library?

Reply:
Yes, tested and is working just fine.

Reply:
great that will please Cloudflare